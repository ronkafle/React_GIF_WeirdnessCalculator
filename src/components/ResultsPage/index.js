import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { Container, Row, Col, Card, CardColumns, Button } from 'react-bootstrap';
import { SEARCH } from '../../actions/search';
import { FAVORITES } from '../../actions/favorites';

const ResultsPage = (props) => {
  const favorites = useSelector(state => state.favorites);
  const dispatch = useDispatch();
  const history = useHistory();

  if (Object.keys(favorites).length < 1) {
    history.replace('/');
  }

  const startOver = () => {
    dispatch({ type: SEARCH.START_OVER });
    dispatch({ type: FAVORITES.START_OVER });
  };

  const score = Object.keys(favorites).reduce((sum, fav) => sum + Number(favorites[fav].weirdness), 0) / 5;

  return (
    <div className="results-page">
      <h3>{`You scored an ${Math.round(score)} out of 10 on the weirdness scale!`}</h3>
      <strong>The GIFs you liked</strong><br />
      <div className="card-wrapper">
        {Object.keys(favorites).map((fav, i) => (
          <Card key={i}>
            <Card.Img variant="top" src={favorites[fav].result.images.original.url} />
            <Card.Body>{favorites[fav].result.title}</Card.Body>
            <Card.Footer>{favorites[fav].weirdness}/10</Card.Footer>
          </Card>
        ))}
      </div>
      <Button onClick={startOver}>
        <i class="fas fa-redo"></i> Start Over
      </Button>
    </div>
  );
};

export default ResultsPage;
