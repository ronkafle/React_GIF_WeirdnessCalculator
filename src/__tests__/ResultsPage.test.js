import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import ResultsPage from '../components/ResultsPage';
import configureStore from "../store";

const store = configureStore();

describe('Component: ResultsPage', () => {
  let component = null;

  beforeEach(() => {
    component = shallow(
      <Provider store={store}>
        <ResultsPage />
      </Provider>
    );
  });
  it('renders without crashing', () => {
    const component = shallow(
      <Provider store={store}>
        <ResultsPage />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
