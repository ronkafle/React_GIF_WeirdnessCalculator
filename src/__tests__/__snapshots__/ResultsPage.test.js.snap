// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ResultsPage renders without crashing 1`] = `ShallowWrapper {}`;
